components/partials/swap/index.vue

```diff
@@ -1473,46 +1536,71 @@ export default Vue.extend({
     },
 
     onSetAmount(quantity: string): void {
-      const { orderTypeBuy, market } = this
+      const {
+        orderTypeBuy,
+        market,
+        fromToken,
+        executionPriceWithoutSlippage,
+        feeRate
+      } = this
 
-      const quantityAsNumber = new BigNumberInBase(Number(quantity))
+      const quantityAsNumber = new BigNumberInBase(quantity)
 
-      const executionPrice = this.calculateExecutionPriceForAmount(
-        quantityAsNumber
-      )
+      this.form.amount = quantity
 
-      const toQuantity = orderTypeBuy
-        ? quantityAsNumber.dividedBy(executionPrice)
-        : executionPrice.times(quantityAsNumber)
+      this.updatePrices()
 
-      this.form.amount = quantity
+      if (!fromToken) {
+        return
+      }
 
-      this.form.toAmount = toQuantity.toFormat(
-        orderTypeBuy ? market?.priceDecimals : market?.quantityDecimals
-      )
+      const decimalPlaces = orderTypeBuy
+        ? market?.priceDecimals
```

Comment:
Let's do an early return if the market is undefined 

```
if (!market) {
    return
}
```

this should actually never happen (market being undefined) so its fine to have an early return